packages/landingpage-new/components/shared/navigationMenu.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState } from 'react';
+import styles from './navigationMenu.module.scss';
+
+export function NavigationMenu() {
+  const [activeSection, setActiveSection] = useState('preparation');
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      (entries) => {
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            setActiveSection(entry.target.id);
+          }
```

Comment:
If we introduce an enum, we could check here whether the `target.id` is part of the given enum. If not, we should print a warning into the console as a hint for fellow developers, that they might have forgotten to at the section to their enum